packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -211,58 +266,87 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget> {
     Size renderSize,
     Offset renderOffset,
   }) {
+    Widget buildEntranceAnimation(BuildContext context, Widget child) {
+      final vertical = properties.isMainAxisVertical;
+      final horizontal = properties.isMainAxisHorizontal;
+      Size size = renderSize;
+      if (childProperties.flexFactor != null) {
+        size = SizeTween(
+          begin: Size(
+            horizontal ? minRenderWidth - entranceMargin : renderSize.width,
+            vertical ? minRenderHeight - entranceMargin : renderSize.height,
+          ),
+          end: renderSize,
+        ).evaluate(expandedEntrance);
+      }
+      return Opacity(
+        opacity: min([allEntrance.value * 5, 1.0]),
+        child: Padding(
+          padding: EdgeInsets.symmetric(
+            horizontal: (renderSize.width - size.width) / 2,
+            vertical: (renderSize.height - size.height) / 2,
+          ),
+          child: child,
+        ),
+      );
+    }
+
     final flexFactor = childProperties.flexFactor;
     return Positioned(
       top: renderOffset.dy,
       left: renderOffset.dx,
       child: InkWell(
-        onTap: _onTap(childProperties),
-        child: Container(
+        onTap: () => _onTap(childProperties),
+        child: SizedBox(
           width: renderSize.width,
           height: renderSize.height,
-          child: WidgetVisualizer(
-            backgroundColor: backgroundColor,
-            title: childProperties.description,
-            borderColor: borderColor,
-            textColor: textColor,
-            child: _visualizeWidthAndHeightWithConstraints(
-              arrowHeadSize: arrowHeadSize,
-              widget: Align(
-                alignment: Alignment.topRight,
-                child: Container(
-                  margin: const EdgeInsets.only(
-                    top: margin,
-                    left: margin,
-                  ),
-                  child: Column(
-                    mainAxisAlignment: MainAxisAlignment.start,
-                    crossAxisAlignment: CrossAxisAlignment.end,
-                    children: <Widget>[
-                      Text(
-                        'flex: $flexFactor',
-                        style: TextStyle(fontWeight: FontWeight.bold),
-                      ),
-                      if (flexFactor == 0 || flexFactor == null)
+          child: AnimatedBuilder(
+            animation: entranceController,
+            builder: buildEntranceAnimation,
+            child: WidgetVisualizer(
```

Comment:
This widget tree is too deeply nested to be easily understood.  Since changing any part of it also changes the indentation of the entire tree, we'll get more merge conflicts working on it together.

A helpful way to manage this is to break the method up into helper methods or to create new helper widget classes.

Reply:
I agree Dave, initially I was going to make a helper widget class for the child but decided not to do that because I was still changing it a lot and I want to avoid 'prop drilling'  into the class and just make use of the scope.

But since the implementation has become more complex, I agree splitting it is helpful.

Should I take this one then? @DaveShuckerow

Reply:
No need now, just pointing it out for future improvements.  I want you focused on making this work off of Flutter master.

It's perfectly OK to write simple code then realize you need to refactor code later on.

Reply:
Alright, will do then, thanks Dave 👍